lending/loan_management/doctype/loan_repayment/loan_repayment.py

```diff
@@ -2603,10 +2603,12 @@ def process_amount_for_loan(
 
 	pending_principal_amount = get_pending_principal_amount(loan, loan_disbursement=loan_disbursement)
 
+	freeze_date = frappe.db.get_value("Loan", loan.name, "freeze_date")
```

Comment:
This db call is not needed, you already have this value in the loan document.

Please add a test case for this scenario in `test_loan_intrest_accrual.py`